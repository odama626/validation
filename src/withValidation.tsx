import * as Core from './coreTypes';
import React from 'react';
import Controller from './controller';
import Form from './form';
import { Wrap } from './wrapper';

export default function(register?) {
  return function(WrappedComponent: React.ComponentType<any>) {
    return class extends React.Component<Core.ExtractProps<typeof WrappedComponent> & { controller: Core.FormController}> {
      controller: Core.FormController;
      validators: any;
      constructor(props) {
        super(props);
        this.controller = new Controller()
        this.validators = {};

        if (register) {
          this.validators = register(validator(this.controller))
        }
      }
      render() {
        return (
          <Form controller={this.controller}>
            <WrappedComponent {...this.props} {...this.validators} controller={this.controller} />
          </Form>
        )
      }
    }
  }
}

export function validator(controller: Core.FormController) {
  return (wrappedComponent: React.ComponentType<Core.FormFieldProps<null>>, validator?: Core.ValidateFunc) => {
      return Wrap(controller, wrappedComponent, validator);
  }
}