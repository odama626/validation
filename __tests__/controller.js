import React from 'react';
import renderer from 'react-test-renderer';
import withValidation, { Form, Controller } from '../lib/validation';

test('withValidation should add controller prop', () => {
  const Validated = withValidation()(({controller}) => (
      <div>{controller.toString()}</div>
    )
  )

  const component = renderer.create(<Validated />);


  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

// test('Controller should get what is set', () => {
//   expect((new Controller()).set('test', { value: 'test', valid: true})).toMatchSnapshot();

// });