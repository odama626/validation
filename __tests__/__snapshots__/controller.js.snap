// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withValidation should add controller prop 1`] = `
<div>
  <div>
    [object Object]
  </div>
</div>
`;
